### ISSUE TITLE
Rustdoc can generate wrong macro_rules! syntax

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust#60543]
Example: https://docs.rs/pkg-version/0.1.0/pkg_version/macro.pkg_version_minor.html

The source code looks like this:
```rust
#[proc_macro_hack]
pub use pkg_version_impl::pkg_version_minor;
```

The rustdoc output looks like this:
```rust
macro_rules! pkg_version_minor {
    #[proc_macro_hack] => { ... };
}
```

Now I'm not entirely sure how `#[proc_macro_hack]` works, but I don't think rustdoc should ever produce... this? The matcher is syntactically invalid, so this should never really happen.

Perhaps related to https://github.com/rust-lang/rust/issues/45939, but I'm not sure.

**Comment by dtolnay** [comment]
Fixed in nightly-2021-07-06, likely by #86282.

**Comment by dtolnay** [comment]
The new rustdoc shows:

```console
macro_rules! pkg_version_minor {
    ($($proc_macro : tt) *) => { ... };
}
```
